nixos/modules/services/networking/nftables.nix

```diff
@@ -110,10 +110,15 @@ in
       wantedBy = [ "multi-user.target" ];
       reloadIfChanged = true;
       serviceConfig = let
+        # Check whether the file is syntactically valid
+        checkedRulesetFile = pkgs.runCommand "nftables-rules-checked"
+          { src = cfg.rulesetFile; }
+          ''${pkgs.nftables}/bin/nft -c -f "$src" && cp $src $out'';
```

Comment:
Does this check attempt to resolve hostnames or network interfaces? This will not work inside sandboxes.

Reply:
Also loading port/service names from /etc/protocol or /etc/services would be a problem.

Reply:
It also requires root:

```
$ cat foo.nft
  table inet filter {
    chain input {
      type filter hook input priority 0;
      ct state established,related accept
      iif lo accept
      # ssh
      tcp dport {22, 22022} accept
    }
  }
$ nft -c -f foo.nft
foo.nft:1:1-2: Error: Could not process rule: Operation not permitted
table inet filter {
^^
foo.nft:2:9-13: Error: Could not process rule: Operation not permitted
  chain input {
        ^^^^^
foo.nft:7:15-25: Error: Could not process rule: Operation not permitted
    tcp dport {22, 22022} accept
              ^^^^^^^^^^^
foo.nft:4:5-39: Error: Could not process rule: Operation not permitted
    ct state established,related accept
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
foo.nft:5:5-17: Error: Could not process rule: Operation not permitted
    iif lo accept
    ^^^^^^^^^^^^^
foo.nft:7:5-32: Error: Could not process rule: Operation not permitted
    tcp dport {22, 22022} accept
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Okay, that really seems to be a deal-breaker. I'll check whether these problems can be avoided somehow. If not, we should do the check in the systemd start script at least, as trying to load a broken ruleset is not very elegant.

Reply:
I had a look at the source code in the current state I don't how this can work without root, which we not have at build time.

Reply:
> Does this check attempt to resolve hostnames or network interfaces? This will not work inside sandboxes.

> Also loading port/service names from /etc/protocol or /etc/services would be a problem.

I'd say that this isn't really an issue, as long as we would provide an option to disable the checks. Many people probably don't use these features.

Reply:
> Okay, that really seems to be a deal-breaker.

It is. I could look into creating some `nft -c` alternative, that doesn't resolve protocols, domains and require a netlink connection, but this will have to wait for a few months. I'll reopen this PR as soon as I've figured something out.

Please excuse the broken PR. :(